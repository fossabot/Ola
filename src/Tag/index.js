import React from 'react'
import {default as PT} from 'prop-types'
import cx from 'classnames'

const Tag = ({ variant, extraClass, children }) =>
  <span className={cx('ola_tag', variant && `is-${variant}`, extraClass)}>
    {children}
  </span>

Tag.defaultProps = {
  variant: null
}

Tag.propTypes = {
  /** Tag variants */
  variant: PT.oneOf(['invert', 'error', 'pro', 'success', 'warning']),
  /** Extra className */
  extraClass: PT.string,
  /** Childen nodes */
  children: PT.oneOfType([
    PT.string,
    PT.arrayOf(PT.node),
    PT.node
  ]).isRequired
}

export { Tag }
