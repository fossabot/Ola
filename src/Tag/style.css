.ola_tag {
    --color: var(--gray);
    --background: var(--gray-xlight);
    --padding: var(--size-3);

    display: inline-flex;
    align-items: center;
    white-space: nowrap;
    color: var(--color);
    background-color: var(--background);
    padding: 0 var(--padding);
    border-radius: var(--radius-small);
    border: solid 1px var(--background);
    vertical-align: baseline;
    cursor: default;
    max-width: 100%;
    letter-spacing: 0;

    & > span {
        text-overflow: ellipsis;
        overflow: hidden;
        min-width: 0;
    }

    &.is-success {
        --color: var(--white);
        --background: var(--success);
    }

    &.is-error {
        --color: var(--white);
        --background: var(--error);
    }

    &.is-warning {
        --color: var(--white);
        --background: var(--warning);
    }

    &.is-invert {
        --color: var(--white);
        --background: var(--gray);
    }

    &.is-info {
      --color: var(--white);
      --background: var(--info);
  }

    &.is-pro {
        --color: var(--white);
        --background: var(--pro);
    }

    &.is-small {
        height: var(--size-6);
        font: var(--font-caption);
        font-weight: var(--bold);
    }

    &.is-medium {
        height: var(--size-7);
        font: var(--font-callout);
        font-weight: var(--bold);
    }

    &.is-big {
        --padding: var(--size-4);
        height: var(--size-8);
        font: var(--font-body);
        font-weight: var(--bold);
    }

    & > .ola_buttonIcon {
        margin-right: calc(var(--padding) * -1);
        margin-left: var(--size-3);
        --color: currentColor;
        flex: 0 0 auto;
        height: 100%;

        & .ola_icon {
            border-radius: 0 calc(var(--radius-small) - 1px) calc(var(--radius-small) - 1px) 0;
            padding-top: 0;
            padding-bottom: 0;
        }
    }
}
