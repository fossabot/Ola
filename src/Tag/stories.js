import React from 'react'
import { storiesOf } from '@storybook/react'

import Tag from './'
import ButtonIcon from '../ButtonIcon'
import Icon from '../Icon'

storiesOf('Tag')
  .add('Default', () => (
    <figure><Tag>Default Tag</Tag></figure>
  ))
  .add('Invert', () => (
    <figure><Tag variant='invert'>Invert Tag</Tag></figure>
  ))
  .add('Success', () => (
    <figure><Tag variant='success'>Sucess Tag</Tag></figure>
  ))
  .add('Warning', () => (
    <figure><Tag variant='warning'>Warning Tag</Tag></figure>
  ))
  .add('Error', () => (
    <figure><Tag variant='error'>Error Tag</Tag></figure>
  ))
  .add('Pro', () => (
    <figure><Tag variant='pro'>Pro Tag</Tag></figure>
  ))
  .add('With a ButtonIcon', () => (
    <figure>
      <Tag variant='invert'>
        Pro Tag
        <ButtonIcon><Icon size="small" name="close" /></ButtonIcon>
      </Tag>
    </figure>
  ))
