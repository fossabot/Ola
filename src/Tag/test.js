import React from 'react'
import Tag from './'
import renderer from 'react-test-renderer'

it('Default Tag', () => {
  const tree = renderer.create(<Tag>Test</Tag>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Tag with invert variant', () => {
  const tree = renderer.create(<Tag variant="invert">Test</Tag>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Tag with error variant', () => {
  const tree = renderer.create(<Tag variant="error">Test</Tag>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Tag with pro variant', () => {
  const tree = renderer.create(<Tag variant="pro">Test</Tag>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Tag with success variant', () => {
  const tree = renderer.create(<Tag variant="success">Test</Tag>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Tag with warning variant', () => {
  const tree = renderer.create(<Tag variant="warning">Test</Tag>).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Tag with className', () => {
  const tree = renderer.create(<Tag className="className">Test</Tag>).toJSON()
  expect(tree).toMatchSnapshot()
})
