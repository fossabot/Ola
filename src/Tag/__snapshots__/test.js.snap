// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Tag 1`] = `
<span
  className="ola_tag is-small"
>
  Test
</span>
`;

exports[`Tag with error variant 1`] = `
<span
  className="ola_tag is-error is-small"
>
  Test
</span>
`;

exports[`Tag with extraClass 1`] = `
<span
  className="ola_tag is-small test"
>
  Test
</span>
`;

exports[`Tag with invert variant 1`] = `
<span
  className="ola_tag is-invert is-small"
>
  Test
</span>
`;

exports[`Tag with pro variant 1`] = `
<span
  className="ola_tag is-pro is-small"
>
  Test
</span>
`;

exports[`Tag with success variant 1`] = `
<span
  className="ola_tag is-success is-small"
>
  Test
</span>
`;

exports[`Tag with warning variant 1`] = `
<span
  className="ola_tag is-warning is-small"
>
  Test
</span>
`;
